core/unpack/unpacker_test.go

```diff
@@ -0,0 +1,86 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package unpack
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/opencontainers/go-digest"
+	"github.com/opencontainers/image-spec/identity"
+)
+
+func generateRandomDiffIDs(num int) []digest.Digest {
+	diffIDs := make([]digest.Digest, num)
+	for i := 0; i < num; i++ {
+		diffIDs[i] = digest.FromString(fmt.Sprintf("sha256:%d", i))
+	}
+	return diffIDs
+}
+
+func BenchmarkUnpackWithChainID(b *testing.B) {
+	// This simulates the old way of repeatedly calculating per-layer chainID
+	// as we unpack every layers, by calling `identity.ChainID`.
+	unpackWithChainID := func(diffIDs []digest.Digest) {
+		var chain []digest.Digest
+		for i := 0; i < len(diffIDs); i++ {
+			_ = identity.ChainID(chain) // parent layer chainID
+			chain = append(chain, diffIDs[i])
+			_ = identity.ChainID(chain).String() // current layer chainID
+		}
+		_ = identity.ChainID(chain).String()
+	}
+
+	numLayers := []int{5, 10, 25, 50}
+	for _, sz := range numLayers {
+		b.Run(fmt.Sprintf("num of layers: %d", sz), func(b *testing.B) {
+			diffIDs := generateRandomDiffIDs(sz)
+			for i := 0; i < b.N; i++ {
+				unpackWithChainID(diffIDs)
+			}
+		})
+	}
+}
+
+func BenchmarkUnpackWithChainIDs(b *testing.B) {
```

Comment:
These 2 benchmarks are mainly to compare the difference between the old and new approaches. We can either keep or remove these tests.